TYPO3.Flow/Tests/Unit/Http/Component/TrustedProxiesComponentTest.php

```diff
@@ -339,6 +339,46 @@ public function hostIsNotOverridenIfTheHeaderIsNotTrusted()
     }
 
     /**
+     * @test
+     */
+    public function hostIsOverridenIfTheHeaderIsTrusted()
+    {
+        $request = Request::create(new Uri('http://acme.com'), 'GET', array(), array(), array('HTTP_X_FORWARDED_HOST' => 'neos.io'));
+        $trustedRequest = $this->callWithRequest($request);
+        $this->assertEquals('neos.io', $trustedRequest->getUri()->getHost());
+    }
+
+    /**
+     * @test
+     */
+    public function portIsOverridenIfTheHostHeaderContainsPort()
+    {
+        $request = Request::create(new Uri('http://acme.com'), 'GET', array(), array(), array('HTTP_X_FORWARDED_HOST' => 'neos.io:443'));
+        $trustedRequest = $this->callWithRequest($request);
+        $this->assertEquals(443, $trustedRequest->getUri()->getPort());
+    }
+
+    /**
+     * @test
+     */
+    public function portIsOverridenIfTheHostHeaderContainsPortAlsoIfProtocolHeaderIsSet()
+    {
+        $request = Request::create(new Uri('http://acme.com'), 'GET', array(), array(), array('HTTP_X_FORWARDED_HOST' => 'neos.io:443', 'HTTP_X_FORWARDED_PROTO' => 'http'));
+        $trustedRequest = $this->callWithRequest($request);
+        $this->assertEquals(443, $trustedRequest->getUri()->getPort());
+    }
+
+    /**
+     * @test
+     */
+    public function portFromHostHeaderIsOverriddenByPortHeader()
```

Comment:
Some more substantial feedback: Is that really the intended behavior that the port header overrides a port from the host header?
(not a rhetoric question, I'm sure that makes sense but I wonder if that's defined somewhere)